packages/models/src/store/store.ts

```diff
@@ -72,22 +146,184 @@ export class Store<
-      throw new NonExistentException(paths.join('.'))
+      throw new NonExistentException(`${key}:${paths.join('.')}`)
     }
     return result
   }
 
-  handleCall = (_msg: ActorMessage<MessagePayload<StoreMessage<GetStorageStruct<StructSchema>>>>): void => {
+  private cloneScript<T extends 'lock' | 'type'>(scriptValue: unknown, type: T): GetFullStorageStruct<StructSchema>[T] {
+    if (typeof scriptValue !== 'object' || scriptValue === null) throw new Error()
+    const res = {} as GetFullStorageStruct<StructSchema>[T]
+    if ('args' in scriptValue) {
+      this.assetStorage(this.getStorage([type, 'args']))
+      res['args'] = this.getStorage([type, 'args'])?.clone(scriptValue.args)
+    }
+    if ('codeHash' in scriptValue) {
+      this.assetStorage(this.getStorage([type, 'codeHash']))
+      res['codeHash'] = this.getStorage([type, 'codeHash'])?.clone(scriptValue.codeHash)
+    }
+    return res
+  }
+
+  private updateChainData(key: OutPointString, paths?: StorePath) {
+    if (!paths) {
+      const value = this.get(key)
+      if ('data' in value) {
+        this.updateValueInCell('data', key, value.data)
+      }
+      if ('witness' in value) {
+        this.updateValueInCell('witness', key, value.witness)
+      }
+      if ('lock' in value) {
+        this.updateScriptInCell('lock', key, value.lock)
+      }
+      if ('type' in value) {
+        this.updateScriptInCell('type', key, value.type)
+      }
+    } else {
+      switch (paths[0]) {
+        case 'data':
+        case 'witness':
+          this.updateValueInCell(paths[0], key, this.get(key, [paths[0]]))
+          break
+        case 'lock':
+        case 'type':
+          this.updateScriptInCell(paths[0], key, this.get(key)[paths[0]])
+          break
+        default:
+          break
+      }
+    }
+  }
+
+  private updateScriptInCell(
+    type: 'lock' | 'type',
+    key: OutPointString,
+    newValue: GetFullStorageStruct<StructSchema>['lock'] | GetFullStorageStruct<StructSchema>['type'],
+  ) {
+    if ('args' in newValue) {
+      this.updateValueInCell([type, 'args'], key, newValue.args)
+    }
+    if ('codeHash' in newValue) {
+      this.updateValueInCell([type, 'codeHash'], key, newValue.codeHash)
+    }
+  }
+
+  private updateValueInCell(type: StorageLocation, key: OutPointString, newValue: unknown) {
+    const cellInfo = this.chainData[key]
+    if (!cellInfo) throw new NonExistentCellException(key)
+    if (typeof type === 'string') {
+      const { offset, length, hexString } = this.serializeField(type, newValue)
+      if (type === 'data') {
+        cellInfo.cell.data =
+          cellInfo.cell.data.slice(0, 2 + offset) +
+          hexString.slice(2) +
+          (length ? cellInfo.cell.data.slice(2 + offset + length) : '')
+      } else {
+        cellInfo.witness =
+          cellInfo.witness.slice(0, 2 + offset) +
+          hexString.slice(2) +
+          (length ? cellInfo.witness.slice(2 + offset + length) : '')
+      }
```

Comment:
I would suggest checking `type` explicitly, and throwing an exception if `type` is neither `data` nor `type`.

It's redundant but ease for mental burden

Reply:
I add `type === 'witness'` with `else if`